src/CacheManager.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache;
+
+use FOS\HttpCache\Exception\ExceptionCollection;
+use FOS\HttpCache\Exception\InvalidArgumentException;
+use FOS\HttpCache\Exception\ProxyResponseException;
+use FOS\HttpCache\Exception\ProxyUnreachableException;
+use FOS\HttpCache\Exception\UnsupportedProxyOperationException;
+use FOS\HttpCache\ProxyClient\ProxyClientInterface;
+use FOS\HttpCache\ProxyClient\Invalidation\BanInterface;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeInterface;
+use FOS\HttpCache\ProxyClient\Invalidation\RefreshInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Manages HTTP cache invalidation.
+ *
+ * @author David de Boer <david@driebit.nl>
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class CacheManager
+{
+    /**
+     * @var array
+     */
+    private $handlers;
+
+    /**
+     * @var array
+     */
+    private $handlerFlushRegistry;
+
+    public function __construct($handlers)
+    {
+        $this->handlers = $handlers;
+    }
+
+    /**
+     * Invoke the invalidate on the named cache handler
+     *
+     * @see CacheHandlerInterface::invaidate
+     *
+     * @param string $handlerName
+     * @param mixed $subject
+     * @param array $options
+     */
+    public function invalidate($handlerName, $subject, array $options = array())
```

Comment:
Well the exception would be useful here and would enumerate ALL the 
possibilities instead of the ones which we happen to define an API for.

On 2015-01-04 17:31, David Buchmann wrote:
> In src/CacheManager.php [1]:
>
>> +
>> + public function __construct($handlers)
>> + {
>> + $this->handlers = $handlers;
>> + }
>> +
>> + /**
>> + * Invoke the invalidate on the named cache handler
>> + *
>> + * @see CacheHandlerInterface::invaidate
>> + *
>> + * @param string $handlerName
>> + * @param mixed $subject
>> + * @param array $options
>> + */
>> + public function invalidate($handlerName, $subject, array $options
> = array())
>
> the problem is also with knowing what to call. $handlerName is just a
> variable, whereas invalidateTags or invalidateRoute is explicit.
>
> --
> Reply to this email directly or view it on GitHub [2].
>
> 	*
>
> Links:
> ------
> [1]
> 
> https://github.com/FriendsOfSymfony/FOSHttpCache/pull/153#discussion-diff-22440816
> [2] 
> https://github.com/FriendsOfSymfony/FOSHttpCache/pull/153/files#r22440816